src/Core/Checkout/Cart/CartPersister.php

```diff
@@ -101,6 +102,15 @@ public function delete(string $token, SalesChannelContext $context): void
         $this->connection->delete('`cart`', ['token' => $token]);
     }
 
+    private function isEmptyCart(Cart $cart): bool
+    {
+        return $cart->getLineItems()->count() <= 0
+            && empty($cart->getExtensions())
```

Comment:
why checking for extensions?

I understand that something similar to customerComment or affiliateCode could be implemented via an extension, but if the extensions were used for something completely different, the logic just never works and you can't get it corrected somehow.

I might just throw an event here:

```
function allowDelete(Cart $cart): bool
{
	
	$allow = (
		$cart->getLineItems()->count <= 0
		&&
		$cart->getCustomerComment() === null
		&&
		// ..
	);

	// ....

	$event = new AllowDeleteCartEvent($allow);
	
	// ...

	return $event->allowDelete();
}
```

Reply:
@OliverSkroblin Shouldn't the cart processor instead take care to ensure any such data is deleted? I really don't see a reason to delete carts, if they have any custom data stored. 

Do you have a use case where this makes sense and one could not ensure to remove the extension in the cart processor, if the cart is empty?

I can adjust it easily, if there are use cases for this though

Reply:
@OliverSkroblin However, I would create an event called like PreSaveCartEvent which has an `requiresCartToBeSaved` flag and then move the empty check to four event subscribers:

- line item count > 0
- customer comment !== null
- afilliation code != null
- check for extension of manual shipping charges because I think those should be kept.

any of those would set `requiresCartToBeSaved` which would also trigger `stopEvent`. 

flag / event names might be subject to change.

Reply:
I am going with the event version now as it is more dynamic, though I still think it is unnecessary.